lib/Sema/ConstraintSystem.cpp

```diff
@@ -789,10 +789,16 @@ ConstraintSystem::getPackElementEnvironment(ConstraintLocator *locator,
       shapeClass->mapTypeOutOfContext()->getCanonicalType());
 
   auto &ctx = getASTContext();
+  GenericEnvironment *outerGenericEnvironment = PackElementGenericEnvironments.empty() ? nullptr : PackElementGenericEnvironments.back();
+
+  auto genericSignature = outerGenericEnvironment ?
+                  outerGenericEnvironment->getGenericSignature().getCanonicalSignature() :
+                  DC->getGenericSignatureOfContext().getCanonicalSignature();
+
   auto elementSig = ctx.getOpenedElementSignature(
-      DC->getGenericSignatureOfContext().getCanonicalSignature(), shapeParam);
+      genericSignature, shapeParam);
   auto *contextEnv = DC->getGenericEnvironmentOfContext();
```

Comment:
Instead of dealing with `outerGenericEnvironment` why can't we just do:

```
 auto *contextEnv = PackElementGenericEnvironments.empty() ? PackElementGenericEnvironments.back() : DC->getGenericEnvironmentOfContext();
```

Move `elementSig` down and use `contextEnv.getGenericSignature()...` for it and for `contextSubs` as well?

Reply:
We absolutely can! Thank you for pointing this out🙂